CONTRIBUTING.md

```diff
@@ -1,7 +1,332 @@
+
+```
+vmdfmt -w <filepath>
+```
+
+To format the entire mdbook from the repository root, outputting a list of files
+modified, run:
+
+```
+vmdfmt -w -l <filepath>
+```
+
+`vmdfmt` is used by the void-docs repository's `check.sh` script, which must be
+[run locally before submitting a pull request](#making-changes).
+
+### Commands
+
+Command code-blocks should start with a `#` or `$` character, indicating whether
+the command should be run as `root` or a regular user, respectively.
+
+For example:
+
+```
+# vi /etc/fstab
+```
+
+and not:
+
+```
+$ sudo vi /etc/fstab
+```
+
+and also not:
+
+```
+vi /etc/fstab
+```
+
+Command code-blocks should be introduced with a colon (':'), i.e.:
+
+> For example:
+> 
+> `$ ls -l`
+
+#### Placeholders
+
+Placeholders indicate where the user should substitute the appropriate
+information. They should use angle brackets (`<` and `>`) and contain only
+lower-case text, with words separated by underscores. For example:
+
+```
+# ln -s /etc/sv/<service_name> /var/service/
+```
+
+and not:
+
+```
+# ln -s /etc/sv/[SERVICENAME] /var/service/
+```
+
+### Links
+
+Link text should not include sentence-level punctuation. For example:
+
+```
+[Visit this site](https://example.org).
+```
+
+and not:
+
+```
+[Visit this site.](https://example.org)
+```
+
+#### Internal links
+
+Links to other sections of the Handbook must be relative. For example:
+
+```
+[example](./example.md#heading-text)
+```
+
+and not:
+
+```
+[example](example.md#heading-text)
+```
+
+When referring literally to a Handbook section, the section title should be
+placed in double-quotes. Otherwise, double-quotes are not required. For example:
+
+```
+For more information, please read the "[Power Management](./power-management.md)" section.
+```
+
+and
+
+```
+Void provides facilities to assist with [power management](./power-management.md).
+```
+
+#### Man Page Links
+
+The first reference to a command or man page must be a link to the relevant man
+page on `https://man.voidlinux.org/`.
+
+The link text must contain the title section number in parenthesis, and contain
+no formatting. For example: [man(1)](https://man.voidlinux.org/man.1), not
+[`man(1)`](https://man.voidlinux.org/man.1).
+
+#### Auto Links
+
+Auto links (links with the same title as URL) should use the following notation:
+
+```
+<https://www.example.com/>
+```
+
+They should not be formatted like this:
+
+```
+[https://www.example.com/](https://www.example.com/)
+```
+
+#### Checking links
+
+If you're including new links (either internal or external) in the docs or
+changing the current file structure, you should make use of the
+`mdbook-linkcheck` package:
+
+```
+$ mdbook-linkcheck -s
+```
+
+This will verify all the references, and warn you if there are any issues. If
+any link you're using is correct but generating errors for some reason, you can
+add its domain to the exclude list in `book.toml`, under the
+`[mdbook.linkcheck]` key.
+
```

Comment:
What is the preferred spelling. "you're" or "you are"? I prefer without a contraction.

Reply:
i prefer with. :-) To me, "you are" sounds oddly stilted and over-formal in this context.

Reply:
Fair enough! Then there's a rogue "you are" in the text you might want to hunt down :p